framework/widgets/ActiveField.php

```diff
@@ -363,6 +363,11 @@ public function hint($content, $options = [])
     public function input($type, $options = [])
     {
         $options = array_merge($this->inputOptions, $options);
+
+        if ($this->form->validationStateOn === ActiveForm::VALIDATION_STATE_ON_INPUT && $this->model->hasErrors($this->attribute)) {
+            $options['class'][] = $this->form->errorCssClass;
+        }
```

Comment:
```
} else {
    $options['class'][] = $this->form->validCssClass;
}
```

This would provide the option to set `is-valid` and on error `is-invalid`. 

https://getbootstrap.com/docs/4.0/components/forms/#server-side

Reply:
That's `successCssClass` and it's already there. There's no need to add this class initially.